### PR TITLE
Merge from AK 2.5 to confluentinc/kafka 2.5

### PR DESCRIPTION
**Pull request by rohitshekhar29**
Resolved merge conflict found in automated merge  https://jenkins.confluent.io/job/apache-kafka-test/job/2.5/117/console
Retained conflicting documentation.

### CODE REVIEW

**Comment by dhruvilshah3** [comment]
Can you add a brief description of what the conflicts were and how you resolved them?

**Comment by rohitshekhar29** [comment]
@dhruvilshah3 the conflict was in docs/upgrade.html with additional comments which I retained.

**Comment by rohitshekhar29** [comment]
Looking into why check fails, it works in my setup though

**Comment by rohitshekhar29** [comment]
It looks like its 2.12 specific scala syntax introduced recently in which is causing build failure.
Thanks, @lbradstreet for guiding me on scala support and pointing the same and @andrewegel with his merge process

**Comment by rohitshekhar29** [comment]
@ijuma @lbradstreet @dhruvilshah3 can this be merged. All scala 2.11 issue are resolved.

File: docs/upgrade.html [hunk 1]
```diff
@@ -30,6 +30,19 @@ <h5><a id="upgrade_250_notable" href="#upgrade_250_notable">Notable changes in 2
...
+    <li>Added a new Serde type <code>Void</code> to represent null keys or null values from input topic.</li>
+    <li>Deprecated <code>UsePreviousTimeOnInvalidTimestamp</code> and replaced it with <code>UsePartitionTimeOnInvalidTimeStamp</code>.</li>
+    <li>Improved exactly-once semantics by adding a pending offset fencing mechanism and stronger transactional commit
+        consistency check, which greatly simplifies the implementation of a scalable exactly-once application.
+        We also added a new exactly-once semantics code example under
+        <a href="https://github.com/apache/kafka/tree/2.5/examples">examples</a> folder. Check out
+        <a href="https://cwiki.apache.org/confluence/display/KAFKA/KIP-447%3A+Producer+scalability+for+exactly+once+semantics">KIP-447</a>
+        for the full details.</li>
+    <li>Deprecated <code>KafkaStreams.store(String, QueryableStoreType)</code> and replaced it with <code>KafkaStreams.store(StoreQueryParameters)</code>.</li>
+    <li>Scala 2.11 is no longer supported. See
```
Comment (lbradstreet) [inline docs/upgrade.html:43]: We shouldn't have included this section about scala 2.11 as we still support scala 2.11 is CCS 2.5.

**Comment by lbradstreet** [comment]
@rohitshekhar29 thanks, the scala 2.11 fixes LGTM. I left one comment about the conflict fix in upgrade.html. Once that's fixed I can approve it.

**Comment by rohitshekhar29** [comment]
merged this from the local repo.

**Pull request closed without merging**